syllabus/core/curriculum_base.py

```diff
@@ -190,14 +230,22 @@ def sample(self, k: int = 1) -> Union[List, Any]:
         assert self.num_tasks > 0, "Task space is empty. Please add tasks to the curriculum before sampling."
 
         if self._should_use_startup_sampling():
-            return self._startup_sample()
-
-        # Use list of indices because np.choice does not play nice with tuple tasks
-        # tasks = self.tasks
-        n_tasks = self.num_tasks
-        task_dist = self._sample_distribution()
-        task_idx = np.random.choice(list(range(n_tasks)), size=k, p=task_dist)
-        return task_idx
+            tasks = self._startup_sample(k)
+            # Check if the startup sampling has satisfied the request or if there's no progress (no tasks returned)
+            if len(tasks) < k and len(tasks) > 0:  # Proceed only if we made progress
+                additional_tasks = self.sample(k=k-len(tasks))
+                tasks.extend(additional_tasks) 
+            return tasks
+        else:
+            task_dist = self._sample_distribution()
```

Comment:
```suggestion

        task_dist = self._sample_distribution()
```
Since the previous if clause returns, we don't need to use else here